### ISSUE TITLE
PolyNull receiver type argument ignores nullable upper bound of wildcard type variable

### ISSUE THREAD

**Issue by aosen-xiong** [eisop/checker-framework#1819]
## Summary

A `@PolyNull` receiver type argument ignores the nullable upper bound of a type variable used as a wildcard extends bound.

When the actual receiver type argument is `? extends T` and `T extends @Nullable Object`, `@PolyNull` should instantiate as `@Nullable`. Instead, it is treated as `@NonNull`, so an assignment to `@NonNull Object` is incorrectly accepted.

## Reproducer

```java
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;
import org.checkerframework.checker.nullness.qual.PolyNull;

abstract class PolyNullReceiverTypeArgumentBound<E> {
    <T extends @Nullable Object> void copyOf(
            PolyNullReceiverTypeArgumentBound<? extends T> elements) {
        @Nullable Object nullable = elements.getObject();

        // Expected error, but currently accepted.
        @NonNull Object nonnull = elements.getObject();
    }

    abstract @PolyNull Object getObject(PolyNullReceiverTypeArgumentBound<@PolyNull E> this);
}
```

I think fixing this will also fix #1074.
